rust/lance-linalg/src/distance/cosine.rs

```diff
@@ -89,7 +91,76 @@ where
 
 impl Cosine for BFloat16Type {}
 
-impl Cosine for Float16Type {}
+#[cfg(any(
+    all(target_os = "macos", target_feature = "neon"),
+    all(target_os = "linux", feature = "avx512fp16")
```

Comment:
`avx512fp16` is a feature passed to LLVM but not recgonized by rustc?

```
rustc --print target-features
```

Can we make sure these kernels are actually called?